[Feature] New AfterTemplatesHaveBeenDeterminedEvent (#2262)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Events/Events/Core/TypoScript/AfterTemplatesHaveBeenDeterminedEvent.rst

.. include:: /Includes.rst.txt
.. index:: Events; AfterTemplatesHaveBeenDeterminedEvent
.. _AfterTemplatesHaveBeenDeterminedEvent:

=====================================
AfterTemplatesHaveBeenDeterminedEvent
=====================================

.. versionadded:: 12.0
   This event is a substitution of the
   :php:`$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['Core/TypoScript/TemplateService']['runThroughTemplatesPostProcessing']`
   hook.

> The event :php:`AfterTemplatesHaveBeenDeterminedEvent` can be used

Reviewer: ```suggestion
The event :php:`\TYPO3\CMS\Core\TypoScript\IncludeTree\Event\AfterTemplatesHaveBeenDeterminedEvent` can be used
```

to manipulate :sql:`sys_template` rows. The event receives the list of resolved
:sql:`sys_template` rows and the :php:`\Psr\Http\Message\ServerRequestInterface` and allows manipulating the
:sql:`sys_template` rows array.

> The event is called in backend EXT:tstemplate code, for example in

Author: ```suggestion
The event is called in the code of the :guilabel:`Template` backend module, for example in
```

the :guilabel:`Template Analyzer`, and in the frontend.

> Extensions using the old hook that want to stay compatible with both TYPO3 v11

Reviewer: ```suggestion
Extensions using the old hook that want to stay compatible with TYPO3 v11
```

Avoid double "both" in one sentence.
